lib/gears/object.lua

```diff
@@ -157,6 +157,95 @@ function object.modulename(level)
     return debug.getinfo(level, "S").source:gsub(".*/lib/", ""):gsub("/", "."):gsub("%.lua", "")
 end
 
+local properties = setmetatable({}, { __mode = 'k' })
+
+local function cobj_register(cobj)
+    local fallback = {
+    }
+
+    function fallback:rawset(_, prop, val)
+        fallback[prop] = val
+    end
+
+    function fallback:rawget(_, prop)
+        return fallback[prop]
+    end
+
+    properties[cobj] = fallback
+    return fallback
+end
+
+--- Add the missing properties handler to a CAPI object such as client/tag/screen
+-- Valid args:
+--
+-- * **getter**: A smart getter (handle property getter itself)
+-- * **getter_fallback**: A dumb getter method (don't handle individual property getter)
+-- * **getter_class**: A module with individual property getter/setter
+-- * **getter_prefix**: A special getter prefix (like "get" or "get_" (default))
+-- * **setter**: A smart setter (handle property setter itself)
+-- * **setter_fallback**: A dumb setter method (don't handle individual property setter)
+-- * **setter_class**: A module with individual property getter/setter
+-- * **setter_prefix**: A special setter prefix (like "set" or "set_" (default))
+-- * **auto_emit**: Emit "property::___" automatically (default: false). This is
+--     ignored when setter_fallback is set or a setter is found
```

Comment:
Should/Can this also have some magic to auto add signals for new properties? I don't really know how that should work, though... It's just annoying to have to add the property signal.